oxd-server/src/main/java/org/xdi/oxd/server/op/GetTokensByCodeOperation.java

```diff
@@ -97,8 +97,9 @@ public IOpResponse execute(GetTokensByCodeParams params) throws Exception {
             return opResponse;
         } else {
             LOG.error("Failed to get tokens because response code is: " + response.getScope());
+            throw new HttpException(ErrorResponseCode.BAD_REQUEST_INVALID_CODE);
```

Comment:
This fix is not accurate. If Authorization Server failed and returned to oxd 500 then in this case we will return back to oxd caller `400` with message `'code' is invalid.`, however it's not true. Code can be valid, it's broken AS that doesn't deliver result. We have to be careful to not mis-lead end-user.

The only thing we can do right now is to check what status code is returned from AS. If it's `400` and we can translate it back to oxd client and return `400` but not for other cases.

Please add tests, so we will see it really works.